### DISCUSSION TITLE
Predicting/Knowing Through thickness location of shell element integration points.

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#21735] [Q&A Modules: Solid mechanics] [answered]
I'm attempting to do some work with shell elements in MOOSE, and as part of this I need to extract curvature from the shell elements. To do this, I have been attempting to use the strain on the upper and lower integration points, however this is not giving me the correct result. I believe I have tracked down the issue to a detail mentioned in a comment on an example file:
``` 
# The stress_yy for the four elements at z = -0.57735 * (t/2) (first qp below mid-surface of shell) are:
# 3031.089 Pa, 2165.064 Pa, 1299.038 Pa and 433.0127 Pa. 
```
In file Beam_Bending_Moment_AD.i
This comment, and the way the stress is extracted by the auxkernel, indicates that in the second order case, the lower integration point is not located at -1/2 thickness, but instead -0.57735*1/2thickness. My question(s):
- why is this the location of the integration point?
- Where will the integration point be located in different through thickness orders with different numbers of integration points? For example, with fourth order through thickness there is one point below the midplane, one at the midplane and one above the midplane, but where are the above and below ones located?
If these values aren't known/communicated to users then it is impossible to know what the stress outputs given by the elements mean. If you take stress values at 0.57735*0.5 of your thickness without realising instead of the top of the shell, they will be artificially lowered.

**Comment by Edward-Eth** [comment, marked as answer]
@GiudGiud @neuphris any thoughts on this?

**Reply by GiudGiud** [reply]
For the first question, this is determined by the quadrature rule used.

I m not very familiar with this, maybe @recuero will know who to hit up for this

**Reply by recuero** [reply]
@neuphris

**Reply by neuphris** [reply]
Like @GiudGiud  mentioned, these integration points come from the Gauss quadrature points. For the second order, the quadrature points are +0.57735 and -0.57735. If you assume the thickness to be 't' with bottom as '0', you can transform the first qp (xi) from the parent domain [-1, 1] to  physical domain [a,b] or [0, t]  as:
x = a/2* (1- xi) + b/2 * (1 - xi) =  0 + t/2 (1-0.57735) = t/2 - t/2 * 0.57735 .
Similarly you can find the location of the quadrature points on the physical domain with higher order.
![Screen Shot 2022-08-08 at 4 09 11 PM](https://user-images.githubusercontent.com/106320412/183523237-ccd76563-f601-4c08-885a-ad9c5c384955.png)

**Reply by Edward-Eth** [reply]
Okay, that makes some sense. Between these points is variation linear or does the interpolation between them change depending on the order (I'm guessing the second). Basically I'm trying to use MOOSE for some shell element FEA and I need shell top and bottom stresses, not stresses at gauss points, so I need a way to extrapolate from the output points to surface stresses (where the stress is maximum in bending and the typical FE output point for a shell). Sorry for pestering, it would be helpful if this information (which is essential to using shells for any serious work) was more easily accessible somewhere.

**Reply by neuphris** [reply]
I have not compared the stresses at the surfaces, but I believe you can take the value of the stress at the Gauss QP and map it to the node using the shape function.
